galsim/chromatic.py

```diff
@@ -131,7 +136,216 @@ def __init__(self, obj):
+    def setupInterpolation(self, waves, oversample_fac=1.):
+        """
+        This method is used as a pre-processing step that can expedite image rendering using objects
+        that have to be built up as sums of GSObjects with different parameters at each wavelength,
+        by interpolating between Images at each wavelength instead of making a more costly
+        instantiation of the relevant GSObject at each value of wavelength at which the bandpass is
+        defined.  This routine does a costly initialization process to build up a grid of images to
+        be used for the interpolation later on.  However, the object can get reused with different
+        bandpasses, so there should not be any need to make many versions of this object, and there
+        is a significant savings each time it is drawn into an image.  As a general rule of thumb,
+        chromatic objects that are separable do not benefit from this particular optimization,
+        whereas those that involve making GSObjects with wavelength-dependent keywords or
+        transformations do benefit from it.  Note that the interpolation scheme is simple linear
+        interpolation in wavelength, and no extrapolation beyond the originally-provided range of
+        wavelengths is permitted.
+
+        The speedup involved in using interpolation depends in part on the bandpass used for
+        rendering (since that determines how many full profile evaluations are involved in rendering
+        the image).  However, for ChromaticAtmosphere with simple profiles like Kolmogorov, the
+        speedup in some simple example cases is roughly a factor of three, whereas for more
+        expensive to render profiles like the ChromaticOpticalPSF, the speedup is more typically a
+        factor of 10-50.
+
+        Achromatic transformations can be applied either before or after setting up interpolation,
+        with the best option depending on the application.  For example, when rendering many times
+        with the same achromatic transformation applied, it is typically advantageous to apply the
+        transformation before setting up the interpolation.  But there is no value in this when
+        applying different achromatic transformation to each object.  Chromatic transformations
+        should be applied before setting up interpolation; attempts to render images of
+        ChromaticObjects with interpolation followed by a chromatic transformation will result in
+        the interpolation being unset and the full calculation being done.
+
+        Because of the clever way that the ChromaticConvolution routine works, convolutions of
+        separable chromatic objects with non-separable ones that use interpolation will still
+        benefit from these optimizations.  For example, a non-separable chromatic PSF that uses
+        interpolation, when convolved with a sum of two galaxy components each with their own SED
```

Comment:
this should probably say "... when convolved with a sum of two separable galaxy components ..."